delimited-iteratee/src/main/scala/net/tixxit/delimited/iteratee/Delimited.scala

```diff
@@ -33,31 +38,13 @@ object Delimited {
    */
   final def inferDelimitedFormat[F[_]](
     bufferSize: Int = DelimitedParser.BufferSize
-  )(implicit F: Applicative[F]): Iteratee[F, String, DelimitedFormat] = {
-    def stepWith(acc: Vector[String], length: Int): Step[F, String, DelimitedFormat] =
-      if (length >= bufferSize) {
-        val chunk = acc.mkString
-        val format = DelimitedFormat.Guess(chunk)
-        Step.done(format)
-      } else {
-        new Step.PureCont[F, String, DelimitedFormat] {
-          final def onEl(e: String): Step[F, String, DelimitedFormat] =
-            stepWith(acc :+ e, length + e.length)
-
-          final def onChunk(h1: String, h2: String, t: Vector[String]): Step[F, String, DelimitedFormat] = {
-            val tLength = t.foldLeft(0) { (n, e) => n + e.length }
-            val newLength = length + h1.length + h2.length + tLength
-            stepWith((acc :+ h1 :+ h2) ++ t, newLength)
-          }
-
-          final def run: F[DelimitedFormat] = {
-            F.pure(DelimitedFormat.Guess(acc.mkString))
-          }
-        }
-      }
-
-    Iteratee.fromStep(stepWith(Vector.empty, 0))
-  }
+  )(implicit F: Monad[F]): Iteratee[F, String, DelimitedFormat] =
+    Enumeratee.scan[F, String, (Vector[String], Int)]((Vector.empty[String], 0)) {
+      case ((acc, length), input) => (acc :+ input, length + input.length)
```

Comment:
do we not want `input :: acc` and use `List[String]` then do `acc.reverse.mkString`. Or maybe in the long run even give the `List[String]` or `Vector[String]` to `DelimitedFormat.Guess` so we don't have to allocate a string just to take it back apart.

Maybe something to look at in the long run, not here. @tixxit ?